packages/test/test-version-utils/src/versionUtils.ts

```diff
@@ -309,31 +312,48 @@ export const loadPackage = async (modulePath: string, pkg: string): Promise<any>
 		for (const key of Object.keys(pkgJson.exports)) {
 			if (!key.startsWith(".")) {
 				throw new Error(
-					"Conditional exports not supported by test-version-utils. Legacy module loading logic needs to be updated.",
+					"Conditional exports are not supported. Legacy module loading logic needs to be updated.",
 				);
 			}
 		}
 		if (typeof pkgJson.exports === "string") {
-			primaryExport = pkgJson.exports;
+			mainEntryPoint = pkgJson.exports;
 		} else {
 			const exp = pkgJson.exports["."];
-			primaryExport =
+			mainEntryPoint =
 				typeof exp === "string"
 					? exp
 					: exp.require !== undefined
 					? exp.require.default
 					: exp.default;
-			if (primaryExport === undefined) {
-				throw new Error(`Package ${pkg} defined subpath exports but no '.' entry.`);
+			if (mainEntryPoint === undefined) {
+				throw new Error(
+					`Package at '${pkgPath}' defined subpath exports but no '.' entry.`,
+				);
 			}
```

Comment:
I like the idea but would rather do it as its own PR in case there's a change in behavior and we have to adapt to it. This PR would just expose the current functionality to other places in the codebase. Thoughts?

Reply:
I think the latest changes address this, can you take a look? However, I added [this comment](https://github.com/microsoft/FluidFramework/pull/20058#discussion_r1523492587) because I'm a bit unsure of what will/should happen as the package versions that define esm exports start to become the ones that are loaded dynamically (maybe in parallel to package's that still don't have that).